README.md

```diff
@@ -57,113 +69,158 @@ Subdirectory README files should follow the same general format as this root fil
+* Servers can have sign-ups that don't require any personal information, so users can use an anonymous alias.
+
+<hr>
+
+## Justifications
+
+* **Squatting:** CAP is a REST protocol built on top of DNS to stop "squatters". Squatting is when users that are the first-adopters of the protocol come in and steal all the valuable aliases ("nike", "coke", "trump", "btc", etc). By requiring that the a domain name is part of an alias, users have to first own the domain which is a system that is already fairly distributed.
+
+* **TXT Records**: There is another protocol out there, OpenAlias, that proposes the usage of domain TXT Records for transmitting alias/address combinations. A lot of good work was done there, but we designed this system because we felt that most defvelopers are more comfortable building out a simple CRUD app with a database. If it is easier for developers to adhere tothe protocol, it will naturally be more distibuted and decentralized.
+
+* **Why no blockchain?** The amount of data being stored on blockchains is a constant struggle, we are always trying to reduce the amount of data that needs to be verified by all nodes in a decentralized network (see the bitcoin scaling problem)Namecoin also never experienced great adoption so we saw no reason to beat that dead horse.
+
+<hr>
+
+## ENDPOINTS
+
+All endpoints (Tier 1 and Tier 2) adhere to RESTful standards and communicate using JSON. 
+
+All endpoints are served at the URL in the following format:
+
+https://opencap.{domain}.{extension}:443/{endpoint}
+
+HTTPS is required for all CAP endpoints, and as is standard the port 443 will be used.
+
+Endpoints are prefixed with Protocol version in case of non-backwards compatibility.
+
+<hr>
+
+## TIER 1 ENDPOINT
 
-### GET /address/{username}/{coin}
+The following GET endpoint is the only endpoint a server needs to support to be CAP 1 compliant. This is because in order for a user to have a CAP alias, wallets just need to know where to look up the address. CAP 1 can be as simple as a static string served from the endpoint.
+
+### GET /v1/address/{ledger_id}/{username}
 
 No Authorization (Public method)
 
-Returns the address of the related username and coin, if they exist.
+Returns the address of the related username and ledger, if it exists on the server.
 
 #### Example Usage
 
 * Bob's decides to send Alice 5 BTC via her alias alice@ogdolo.com
 * Bob's wallet parses the alias into a URL destination and HTTP method
-* alice@ogdolo.com -> GET <https://cap.ogdolo.com/address/alice/btc>
+* BTC_SEGWIT = ledger_id: 1
+* alice@ogdolo.com -> GET <https://opencap.ogdolo.com/address/1/alice>
 * If alice truly has a Bitcoin address hosted on the domain that her alias suggests, the address will be sent back to Bob's wallet with an HTTP 200 Status OK.
 
 ```javascript
 Code: 200
 {
-    "address": "bc1qvw0ytfntx6zs0lfsruem6xwj0mewng523ktatp",
+	"message": "success",

```

Comment:
Not required if we use HTTP status codes

Reply:
The message is there for the text, the status code is just a code

Reply:
What should the message be used for? Either the server returns 200 and an address or 404 if there's no address.

Reply:
500 internal server error, deprecation, etc, etc, just anything they want to say to give a good description of the problem. There will be more things that can potentially go wrong with the other endpoints. It will mostly help devs when they are building clients to see what they are doing wrong.

Reply:
There's no deprecation as we have v1 as part of the path.  
Only two things can happen: Either an address is found or an address is not found. If there's an unexpected error it's not a thing that the wallet developer should care about.

Reply:
Okay, we can remove the message